### PR TITLE
feat: added cookies API

### PR DESCRIPTION
**Pull request by Aleksey28**
## Purpose
Implement cookie API

## Approach
1. Create commands for working with cookies
2. Add preparing arguments for commands
3. Add validation for commands arguments
4. Use testcafe-hammerhead API  to manipulate cookies store
5. Add functional and unit tests

## API methods
```ts
/**
 * Returns cookies associated with the specified cookie objects. If there are no parameters, this method returns all cookies.
 *
 * @param cookies - The cookie objects.
 */
getCookies(...cookies: any): Promise<any>;
/**
 * Returns cookies with the specified names and URL contexts.
 *
 * @param names - The cookie names.
 * @param urls - The URL.
 */
getCookies(names: string | string[], urls?: string | string[]): Promise<any>;
/**
 * Sets the specified cookie objects.
 *
 * @param cookies - The cookie objects.
 */
setCookies(...cookies: any): TestControllerPromise;
/**
 * Sets cookies' name-value pairs to the specified URL contexts.
 *
 * @param nameValueObjects - The cookies' name-value pairs.
 * @param url - The URL.
 */
setCookies(nameValueObjects: Record<string, string> | Record<string, string>[], url?: string): TestControllerPromise;
/**
 * Deletes cookies. If there are no parameters, this method deletes all cookies.
 *
 * @param cookies - The cookies.
 */
deleteCookies(...cookies: any): TestControllerPromise;
/**
 * Deletes cookies associated with the URL contexts. If there are no parameters, this method deletes all cookies.
 *
 * @param names - The cookie names.
 * @param urls - The URL.
 */
deleteCookies(names: string | string[], urls?: string | string[]): TestControllerPromise;
```

## References
[PR in the testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead/pull/2731)
[testcafe-hammerhead-24.5.13.zip with API](https://github.com/DevExpress/testcafe-hammerhead/files/7959114/testcafe-hammerhead-24.5.13.zip)

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

File: src/api/test-controller/index.js [hunk 1]
```diff
@@ -213,6 +218,51 @@ export default class TestController {
...
+                normalizedCookies.push({ name: cookie });
+            else
+                normalizedCookies.push(cookie);
+        });
+
+        return normalizedCookies;
+    }
+
+    [delegatedAPI(GetCookiesCommand.methodName)] (...args) {
+        const urls    = this._getUrls(args);
```
Comment (miherlosev) [inline src/api/test-controller/index.js:246]: I propose to pack the `_getUrls` and `_noramalizeCookies` methods to the single `_prepareCookieArguments` and cover it in the server tests.

File: src/test-run/index.ts [hunk 2]
```diff
@@ -775,6 +783,159 @@ export default class TestRun extends AsyncEventEmitter {
         return consoleMessageCopy[String(this.activeWindowId)];
     }
 
+    private async _findCookiesByApi (urls: { domain: string; path: string }[], key?: string): Promise<(InternalCookie | InternalCookie[])[]> {
```
Comment (miherlosev) [inline src/test-run/index.ts:786]: I propose to pack the cookie-related API to the `CookieManagment` class that will be initialized with the `this.session.cookies`. It allows to cover all functionality in the server tests and reduce the functional tests to 3 items: `SetCookie`, `GetCookie`, and `deleteCookie`.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: src/test-run/index.ts (hunk 2 above)
Comment (Aleksey28) [reply src/test-run/index.ts]: I moved this functionality to the hammerhead and add tests for one there.

File: test/functional/fixtures/api/es-next/cookies/test.js [hunk 3]
```diff
@@ -1,75 +1,25 @@
-describe('[API] Cookies', function () {
-    it('Should get all cookies', function () {
-        return runTests('./testcafe-fixtures/cookies-test.js', 'Should get all cookies');
-    });
+/*eslint-disable*/
+describe.only('[API] Cookies', function () {
```
Comment (miherlosev) [inline test/functional/fixtures/api/es-next/cookies/test.js:2]: Remove the `eslint-disable` comment and the `describe.only` hook.

File: src/api/test-controller/index.js [hunk 4]
```diff
@@ -218,49 +218,37 @@ export default class TestController {
         return this._enqueueCommand(DispatchEventCommand, { selector, eventName, options, relatedTarget: options.relatedTarget });
     }
 
-    _getUrls (args) {
-        const [, urls] = args;
+    _prepareCookieArguments (args, isSetCommand = false) {
+        const urlsArg = castArray(args[1]);
```
Comment (miherlosev) [inline src/api/test-controller/index.js:222]: I guess we should add argument validation for common cases:
```js
await t.setCookie();
await t.getCookie(null);
//etc.
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by miherlosev — APPROVED** [review]

File: src/api/test-controller/index.js (hunk 4 above)
Comment (Aleksey28) [reply src/api/test-controller/index.js:222]: Validation implemented the same way as for another commands of TestController

**Review by AndreyBelym — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -133,8 +133,8 @@
     "semver": "^5.6.0",
     "source-map-support": "^0.5.16",
     "strip-bom": "^2.0.0",
-    "testcafe-browser-tools": "2.0.22",
-    "testcafe-hammerhead": "24.5.13",
+    "testcafe-browser-tools": "2.0.20",
+    "testcafe-hammerhead": "https://github.com/DevExpress/testcafe-hammerhead/files/8022603/testcafe-hammerhead-24.5.13.zip",
```
Comment (AndreyBelym) [inline package.json:137]: Please use latest stable versions

**Pull request merged**